CorsixTH/Lua/objects/doors/swing_door_right.lua

```diff
@@ -39,18 +39,14 @@ function SwingDoor:SwingDoor(hospital, object_type, x, y, direction, etc)
   self.is_master = object_type == object
   self:Door(hospital, object_type, x, y, direction, etc)
   if self.is_master then
-    -- Wait one tick before finding the slave so that we're sure it has been created.
-    local --[[persistable:swing_door_creation]] function callback()
-      local slave_type = "swing_door_left"
-      self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type) or nil
-      self.slave:setAsSlave(self)
-      self.ticks = false
-    end
-    self:setTimer(1, callback)
-    self.ticks = true
+    local slave_type = "swing_door_left"
+    self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type) or nil
+    self.slave:setAsSlave(self)
+    assert(self.slave.master, "Swing doors are not paired at " .. x .. "," .. y)
```

Comment:
Is this assert for testing or have we thoroughly checked that the left door exists first on creation before the right door (in terms of code order)

Reply:
They are always built together, regardless of orientation. I can't cause this assert to trigger, so I expect no one ever reaches it.